libraries/resource_helpers.rb

```diff
@@ -22,10 +22,27 @@ module Helpers
     def data_dir
       case node['platform_family']
       when 'rhel', 'fedora', 'amazon'
-        "/var/lib/pgsql/#{new_resource.version}/data"
+        "/var/lib/pgsql/#{node.run_state['postgresql']['version']}/data"
       when 'debian', 'ubuntu'
-        "/var/lib/postgresql/#{new_resource.version}/main"
+        "/var/lib/postgresql/#{node.run_state['postgresql']['version']}/main"
       end
     end
+
+    def conf_dir
+      case node['platform_family']
+      when 'rhel', 'fedora', 'amazon'
+        "/var/lib/pgsql/#{node.run_state['postgresql']['version']}/data"
+      when 'debian', 'ubuntu'
```

Comment:
no need for ubuntu here. Debian platform family includes ubuntu

Reply:
Agree